homeassistant/components/light/deconz.py

```diff
@@ -0,0 +1,182 @@
+"""
+Support for deCONZ light support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/light/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    DECONZ_DATA, DOMAIN)
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS, ATTR_FLASH, ATTR_COLOR_TEMP, ATTR_EFFECT,
+    ATTR_RGB_COLOR, ATTR_TRANSITION, EFFECT_COLORLOOP, FLASH_LONG, FLASH_SHORT,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP, SUPPORT_EFFECT, SUPPORT_FLASH,
+    SUPPORT_RGB_COLOR, SUPPORT_TRANSITION, SUPPORT_XY_COLOR)
+from homeassistant.core import callback
+from homeassistant.util.color import color_RGB_to_xy
+
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LIGHT_GROUP = 'LightGroup'
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup light platform for Deconz."""
+    if DECONZ_DATA in hass.data:
+        lights = hass.data[DECONZ_DATA].lights
+        groups = hass.data[DECONZ_DATA].groups
+
+    for _, light in lights.items():
```

Comment:
Should probably be inside of the above `if` statement, otherwise `lights` and `groups` will be undefined.

Reply:
They're created as empty dicts so it will work out as expected.